src/main.rs

```diff
@@ -94,7 +96,7 @@ where
 
     let keyboard_events = os_input.keyboard_events;
     let get_open_sockets = os_input.get_open_sockets;
-    let lookup_addr = os_input.lookup_addr;
+    let resolver_factory = os_input.resolver_factory;
```

Comment:
Why are we using a resolver_factory rather than constructing resolver? (I'm sure there's a legit reason, just want to understand :) )

Reply:
So initially I thought about doing that to be able to return a fake resolver in the tests, but I see now that it's not necessary and I could just use a boxed Lookup trait object. In any case, I think I will change this after thinking about your last comment (will reply there :) )